condor/submitter.py

```diff
@@ -0,0 +1,240 @@
+    parser = get_main_parser()
+    args = parser.parse_args()
+    print("Running with the following options:")
+    print(args)
+
+    current_dir = os.path.dirname(os.path.abspath(__file__))
+    base_dir = current_dir.replace("/condor", "")
+
+    if args.remoteRepo is not None:
+        print("Will use a remote path to access BTVNanoCommissioning:", args.remoteRepo)
+    else:
+        print("Tarring BTVNanoCommissioning directory...")
+
+        skip_tar = False
+        if os.path.exists("BTVNanoCommissioning.tar.gz"):
+            user_input = input(
+                "BTVNanoCommissioning.tar.gz already exists, skip the tarring? (y/n): "
+            )
+            if user_input.lower() == "y":
+                skip_tar = True
+            elif user_input.lower() == "n":
+                skip_tar = False
+                os.remove("BTVNanoCommissioning.tar.gz")
+            else:
+                raise Exception("Invalid input, exiting")
+
+        if not skip_tar:
+            make_tarfile(
+                "BTVNanoCommissioning.tar.gz",
+                base_dir,
+                exclude_dirs=["jsonpog-integration", "BTVNanoCommissioning.egg-info"],
+            )
+
+    # Create job dir
+    job_dir = f"jobs_{args.jobName}"
+    if os.path.exists(job_dir):
+        user_input = input("Job directory already exists, overwrite? (y/n): ")
+        if user_input.lower() == "y":
+            shutil.rmtree(job_dir)
+        else:
+            raise Exception("Job exiting...")
+    print(f"Job directory created: {job_dir}")
+    os.mkdir(job_dir)
+    os.mkdir(job_dir + "/log")
+
+    # Store job submission files
+
+    ## store parser arguments
+    with open(os.path.join(job_dir, "arguments.json"), "w") as json_file:
+        json.dump(vars(args), json_file, indent=4)
+
+    ## split the sample json
+    with open(args.samplejson) as f:
+        sample_dict = json.load(f)
+    split_sample_dict = {}
+    counter = 0
+    for sample_name, files in sample_dict.items():
+        for ifile in range(
+            (len(files) + args.condorFileSize - 1) // args.condorFileSize
+        ):
+            split_sample_dict[counter] = {
+                sample_name: files[
+                    ifile * args.condorFileSize : (ifile + 1) * args.condorFileSize
+                ]
+            }
+            counter += 1
+
+    ## store the split sample json file
+    with open(os.path.join(job_dir, "split_samples.json"), "w") as json_file:
+        json.dump(split_sample_dict, json_file, indent=4)
+    ## store the jobnum list (0..jobnum-1)
+    with open(os.path.join(job_dir, "jobnum_list.txt"), "w") as f:
+        f.write("\n".join([str(i) for i in range(counter)]))
+
+    ## store the jdl file
+    jdl_template = """Universe   = vanilla
+Executable = {executable}
+
++ProjectName="cms.org.cern"
+
+Arguments = $(JOBNUM)
+
+requirements = (OpSysAndVer =?= "CentOS7")
+request_cpus = 1
+request_memory = 2000
+use_x509userproxy = true
```

Comment:
Just realize `condor` at lxplus is a bit older. ` x509userproxy = /tmp/x509up_u95636` is still required to specified

Reply:
I also use 9.0.17.. I don't know why if I don't specified `x509userproxy` it could not found in the condor jobs...

Reply:
I also use `9.0.17 ` but not sure why if I don't specify ` x509userproxy` I would run into errors for not finding proxy...

Reply:
ah I realized I had a special setting in lxplus `.bashrc`: 
```
export X509_USER_PROXY=$HOME/.krb5/x509up_u`id -u` 
```

Reply:
might this be the reason? 🤔

Reply:
Then do you have special config for voms-init-proxy? just wondering how the proxy went in `$HOME/.krb5/` while mine usually stays in `$HOME/x509up_xxx`

Reply:
I have something similar to @colizz, in my case, I have `export X509_USER_PROXY=${HOME}/.gridproxy.pem` in my `.bashrc`, my voms-proxy-init command is the usual one though (`voms-proxy-init --voms cms --valid 168:00`), i.e. the fact that the proxy ends up in a specific folder should be solely driven by `export X509_USER_PROXY`

Reply:
Yes, my understanding is the same. This always works for me without more x509 setup on condor jdl (just `use_x509userproxy = true`)